pkg/controller/machinehealthcheck/machinehealthcheck_controller.go

```diff
@@ -170,23 +182,56 @@ func remediate(r *ReconcileMachineHealthCheck, machine *mapiv1.Machine) (reconci
 		return reconcile.Result{}, err
 	}
 
-	if !isHealthy(node) {
-		if unhealthyForTooLong(node) {
-			// delete
-			glog.Infof("machine %s has been unhealthy for too long, deleting", machine.Name)
-			if err := r.client.Delete(context.TODO(), machine); err != nil {
-				glog.Errorf("failed to delete machine %s, requeuing referenced node", machine.Name)
+	cmUnhealthConditions := &corev1.ConfigMap{}
+	cmKey := types.NamespacedName{
+		Name:      healthcheckingv1alpha1.ConfigMapNodeUnhealthConditions,
+		Namespace: r.namespace,
+	}
+	err = r.client.Get(context.TODO(), cmKey, cmUnhealthConditions)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			glog.Warningf("ConfigMap %s not found", healthcheckingv1alpha1.ConfigMapNodeUnhealthConditions)
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+	nodeUnhealthConditions, err := conditions.GetNodeUnhealthyConditions(node, cmUnhealthConditions)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	var result *reconcile.Result
+	for _, c := range nodeUnhealthConditions {
+		nodeCondition := conditions.GetNodeCondition(node, corev1.NodeConditionType(c.Name))
+		if nodeCondition != nil && nodeCondition.Status == corev1.ConditionStatus(c.Status) {
```

Comment:
nit: although it wouldn't give us reusability, I'd personally like to see this `if` sentence in an english readable format for clarity